Xamarin.Forms.Core/FlexLayout.cs

```diff
@@ -0,0 +1,451 @@
+			flexLayout.InvalidateLayout();
+		}
+
+		static void OnJustifyContentPropertyChanged(BindableObject bindable, object oldValue, object newValue)
+		{
+			var flexLayout = bindable as FlexLayout;
+			if (flexLayout._root == null)
+				return;
+			flexLayout._root.JustifyContent = (Flex.Align)(FlexJustify)newValue;
+			flexLayout.InvalidateLayout();
+		}
+
+		static void OnAlignContentPropertyChanged(BindableObject bindable, object oldValue, object newValue)
+		{
+			var flexLayout = bindable as FlexLayout;
+			if (flexLayout._root == null)
+				return;
+			flexLayout._root.AlignContent = (Flex.Align)(FlexAlignContent)newValue;
+			flexLayout.InvalidateLayout();
+		}
+
+		static void OnAlignItemsPropertyChanged(BindableObject bindable, object oldValue, object newValue)
+		{
+			var flexLayout = (FlexLayout)bindable;
+			if (flexLayout._root == null)
+				return;
+			flexLayout._root.AlignItems = (Flex.Align)(FlexAlignItems)newValue;
+			flexLayout.InvalidateLayout();
+		}
+
+		static void OnPositionPropertyChanged(BindableObject bindable, object oldValue, object newValue)
+		{
+			var flexLayout = (FlexLayout)bindable;
+			if (flexLayout._root == null)
+				return;
+			flexLayout._root.Position = (Flex.Position)(FlexPosition)newValue;
+			flexLayout.InvalidateLayout();
+		}
+
+		static void OnWrapPropertyChanged(BindableObject bindable, object oldValue, object newValue)
+		{
+			var flexLayout = bindable as FlexLayout;
+			if (flexLayout._root == null)
+				return;
+			flexLayout._root.Wrap = (Flex.Wrap)(FlexWrap)newValue;
+			flexLayout.InvalidateLayout();
+		}
+
+		Flex.Item _root;
+		//this should only be used in unitTests. layout creation will normally happen on OnParentSet
+		internal override void OnIsPlatformEnabledChanged()
+		{
+			base.OnIsPlatformEnabledChanged();
+			if (IsPlatformEnabled && _root == null)
+				PopulateLayout();
+			else if(!IsPlatformEnabled && _root != null)
+				ClearLayout();
+		}
+
+		protected override void OnParentSet()
+		{
+			base.OnParentSet();
+			if (Parent != null && _root == null)
+				PopulateLayout();
+			else if (Parent == null && _root != null)
+				ClearLayout();
+		}
+
+		void PopulateLayout()
+		{
+			InitLayoutProperties(_root = new Flex.Item());
+			foreach (var child in Children)
+				AddChild(child);
+		}
+
+		void InitLayoutProperties(Flex.Item item)
+		{
+			var values = GetValues(AlignContentProperty,
+								   AlignItemsProperty,
+								   DirectionProperty,
+								   JustifyContentProperty,
+								   WrapProperty);
+			item.AlignContent = (Flex.Align)(FlexAlignContent)values[0];
+			item.AlignItems = (Flex.Align)(FlexAlignItems)values[1];
+			item.Direction = (Flex.Direction)(FlexDirection)values[2];
+			item.JustifyContent = (Flex.Align)(FlexJustify)values[3];
+			item.Wrap = (Flex.Wrap)(FlexWrap)values[4];
+		}
+
+		void ClearLayout()
+		{
+			foreach (var child in Children)
+				RemoveChild(child);
+			//_root.Dispose();
```

Comment:
Is this intentional?

Reply:
it should go, the managed engine doesn't need to be disposed.

Reply:
fixed